__tests__/pet.test.js

```diff
@@ -0,0 +1,158 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () => {
+    it('returns an object', () => {
+        expect(new Pet('Fido')).toBeInstanceOf(Object);
+    });
+    it('sets the name property', () => {
+        const pet = new Pet('Fido');
+        expect(pet.name).toEqual('Fido');
+        const pet2 = new Pet('Eddy');
+        expect(pet2.name).toEqual('Eddy');
+    });
+    it('has an initial age of 0', () => {
+        const pet = new Pet('Fido');
+        expect(pet.age).toEqual(0);
+    });
+});
+
+describe('growUp', () => {
+    it('increments the age by 1', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.age).toEqual(1);
+        pet.growUp();
+        expect(pet.age).toEqual(2);
+    });
+    it('increments hunger by 5 for every 1 age', () => {
+        const pet2 = new Pet('Fido');
+        expect(pet2.hunger).toEqual(0);
+        pet2.growUp();
+        expect(pet2.hunger).toEqual(5);
+    })
+    it('decreases fitness by 3 for every 1 age', () => {
+        const pet3 = new Pet('Fido');
+        expect(pet3.fitness).toEqual(10);
+        pet3.growUp();
+        expect(pet3.fitness).toEqual(7);
+        pet3.growUp();
+        expect(pet3.fitness).toEqual(4);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+        pet.age = 31;
+        expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+    });
+
+});
+
+describe('walk', () => {
+    it('increases fitness by 4', () => {
+        const pet = new Pet ('fido');
+        pet.fitness = 4;
+        pet.walk();
+        expect(pet.fitness).toEqual(8);
+    });
+    it('increases fitness to a maximum of 10', () => {
+        const pet = new Pet ('fido');
+        pet.fitness = 8;
+        pet.walk();
+        expect(pet.fitness).toEqual(10);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+        pet.hunger = 11;
+        expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+    });
+});
+
+describe('feed', () => {
+    it('decreases hunger by 3', () => {
+        const newPet = new Pet ('Lily');
+        newPet.growUp();
+        newPet.feed();
+        expect(newPet.hunger).toEqual(2);
+    });
+    it('decreases hunger to a minimum of 0', () => {
+        const pet = new Pet ('Eddy')
+        pet.hunger = 2;
+        pet.feed();
+        expect(pet.hunger).toEqual(0);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+        pet.age = 30;
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+});
+
+describe('checkUp', () => {
```

Comment:
Inside each `it` block you dont need to name each instance of `pet` differently, as they are all in their own scope. The first line of each of these tests could be `const pet = new Pet ('somename');` and it would work just fine